cloudinit/sources/DataSourceOVF.py

```diff
@@ -684,4 +740,73 @@ def _raise_error_status(prefix, error, event, config_file, conf):
     util.del_dir(os.path.dirname(config_file))
     raise error
 
+
+def load_cloudinit_data(metaPath, userPath):
+    """
+    Load the cloud-init meta data, user data, cfg and network from the
+    given files
+    """
+    LOG.debug('load meta data from: %s: user data from: %s',
+              metaPath, userPath)
+    md = {}
+    cfg = {}
+    ud = None
+    network = None
+
+    md = load(util.load_file(metaPath))
+
+    if 'network' in md:
+        network = md['network']
+        del md['network']
+
+    if userPath:
+        ud = util.load_file(userPath).replace("\r", "")
+    return md, ud, cfg, network
+
+
+def load(data):
+    '''
+    The meta data could be JSON or YAML. Since YAML is a strict superset of
+    JSON, we will unmarshal the data as YAML. If data is None then a new
+    dictionary is returned.
+    '''
+    if not data:
+        return {}
+    return safeyaml.load(data)
+
+
+def collect_imc_files(cust_conf):
+    '''
+    collect all the other imc files. Since these files are copied
+    before cust.cfg is copied, check if they exist or not directly.
```

Comment:
I'm not quite sure about the doctstring meaning with "not directly"?

Also from the code it looks like metadata/userdata are mutually exclusive from nics.txt. If this is true, let's qualify the docstring to say either metadata/userdata or nics.txt must be present.

Reply:
metadata/userdata must be present if they are specified in customization configuration. In this case, nics.txt is not needed. For other case, we should try to load nics.txt (optional)